HW02_B_Mimic_starter.Rmd

```diff
@@ -1,99 +1,125 @@
+#theme_bw() is best theme (IMO)
+
+#for graph 3:
+library("ggrepel")
+
+
+```
+
+## HW02 Part B
+
+For this part of the HW, the goal is to try to recreate the graphs I make from scratch. I will only provide the MD, not the actual code I used to create it besides which data I use to create it. The rest will be up to you. 
+
+Try for all 4, but if you are struggling don't worry about it. Try your best for each, if you don't get everything that's what the peer-review is for. :smile:
+
+
+### Graph 1
+```{r, echo = T, include = TRUE}
+data("diamonds")
+#hint think about the *position* the bars are in...
+```
+
+
+Using the diamonds dataset, make this graph:
+```{r graph1 code, echo=FALSE}
+ggplot(diamonds, aes(x= cut)) + 
+  geom_histogram(aes(fill=clarity),position = "dodge", stat='count')  +  
+#'count' automatically computes the number in each category
+  labs(x = "Diamond Cut", y= "Number of Diamonds", title="My Diamond Collection", subtitle = "Boxplot representing the number of diamonds in my diamond collection by \ntype of cut quality and clarity of diamond" ) +
+  theme(plot.title = element_text(hjust = 0.5)) + 
+#geom_text(aes(x="Premium", y=4500), label="My Best Diamonds,\n of course") also works but slow hehe
+annotate("text", x = "Premium", y = 4500, label ="My Best Diamonds,\n of course")
+#\n creates new line
+```
+
+### Graph 2
+```{r, echo = T, include = TRUE}
+data("iris")
+```
+
+Using the iris dataset, make this graph:
+```{r graph 2 code, echo=FALSE}
+
+iris$Species <- factor(iris$Species, levels = c("versicolor","setosa","virginica"))
+#changing order of factor to un-alphabetical (non-default)
+ggplot(iris, aes(x= Sepal.Length, y= Petal.Length )) +
+  geom_point(aes(color = Species, shape = Species)) + 
+  geom_smooth(formula = y~x , method = 'lm', se= F, color= 'black') + facet_wrap(~Species, scales="free_y")
+#geom_smooth 'lm' fits linear model, se=F does not print standard error
+#facet_wrap breaks up plot, a diff plot for each species, scale=free y means each species has its own y scale
+#dots and triangles are switched???
+```
+
+
+### Graph 3
+You'll need the information in this first box to create the graph
+```{r, echo = TRUE}
+data("mpg")
+corvette <- mpg[mpg$model == "corvette",] #[only select the rows where the model= tcorvette within the mpg dataset], which is being put (<-)into a vector
+#install
+require("ggrepel") #ggrepel provides geoms for ggplot to repel overlapping text labels. useful for making text annotations better, hint hint
+set.seed(42)
+```
+
+Now using the mpg dataset and the corvette dataset, make this graph:
+
+```{r graph 3 code}
+library(ggrepel)
+
+ggplot(mpg, aes(x=displ, y=hwy))+ 
+  geom_point() + 
+  geom_text_repel(corvette, mapping = aes(x= displ, y=hwy, label =   paste("Corvette,", year))) + 
+  geom_point(data=corvette, color = 'blue') + 
+  labs(title = "Corvettes are a bit of an outlier")
+#mapping calls displ and hwy from corvette specifically, not mpg's displ and hwy
```

Comment:
Hi Monty, I think calling corvette already did the job, so no need for mapping calls display and hwy?
Mine was without them, and seemed worked ok?
geom_text_repel(data = corvette, label = paste("Corvette,",corvette$year)